wren-ui/src/apollo/server/adaptors/wrenAIAdaptor.ts

```diff
@@ -318,12 +332,123 @@ export class WrenAIAdaptor implements IWrenAIAdaptor {
+        `${this.wrenAIBaseEndpoint}/v1/charts`,
+        input,
+      );
+      return { queryId: res.data.query_id };
+    } catch (err: any) {
+      logger.debug(`Got error when creating chart: ${getAIServiceError(err)}`);
+      throw err;
+    }
+  }
+
+  public async getChartResult(queryId: string): Promise<ChartResult> {
+    try {
+      const res = await axios.get(
+        `${this.wrenAIBaseEndpoint}/v1/charts/${queryId}`,
+      );
+      return this.transformChartResult(res.data);
+    } catch (err: any) {
+      logger.debug(
+        `Got error when getting chart result: ${getAIServiceError(err)}`,
+      );
+      throw err;
+    }
+  }
+
+  public async cancelChart(queryId: string): Promise<void> {
+    try {
+      await axios.patch(`${this.wrenAIBaseEndpoint}/v1/charts/${queryId}`, {
+        status: 'stopped',
+      });
+    } catch (err: any) {
+      logger.debug(`Got error when canceling chart: ${getAIServiceError(err)}`);
+      throw err;
+    }
+  }
+
+  public async adjustChart(
+    input: ChartAdjustmentInput,
+  ): Promise<AsyncQueryResponse> {
+    try {
+      const res = await axios.post(
+        `${this.wrenAIBaseEndpoint}/v1/chart-adjustments`,
+        this.transformChartAdjustmentInput(input),
+      );
+      return { queryId: res.data.query_id };
+    } catch (err: any) {
+      logger.debug(`Got error when adjusting chart: ${getAIServiceError(err)}`);
+      throw err;
+    }
+  }
+
+  public async getChartAdjustmentResult(queryId: string): Promise<ChartResult> {
+    try {
+      const res = await axios.get(
+        `${this.wrenAIBaseEndpoint}/v1/chart-adjustments/${queryId}`,
+      );
+      return this.transformChartResult(res.data);
+    } catch (err: any) {
+      logger.debug(
+        `Got error when getting chart adjustment result: ${getAIServiceError(err)}`,
       );
       throw err;
     }
   }
 
+  public async cancelChartAdjustment(queryId: string): Promise<void> {
+    try {
+      await axios.patch(
+        `${this.wrenAIBaseEndpoint}/v1/chart-adjustments/${queryId}`,
+        {
+          status: 'stopped',
+        },
+      );
+    } catch (err: any) {
+      logger.debug(
+        `Got error when canceling chart adjustment: ${getAIServiceError(err)}`,
+      );
+      throw err;
+    }
+  }
+
+  private transformChartAdjustmentInput(input: ChartAdjustmentInput) {
+    const { query, sql, adjustmentOption, chartSchema } = input;
+    return {
+      query,
+      sql,
+      adjustment_option: {
+        chart_type: adjustmentOption.chartType.toLowerCase(),
+        x_axis: adjustmentOption.xAxis,
+        y_axis: adjustmentOption.yAxis,
+        x_offset: adjustmentOption.xOffset,
+        color: adjustmentOption.color,
+        theta: adjustmentOption.theta,
+      },
+      chart_schema: chartSchema,
+    };
+  }
+
+  private transformChartResult(body: any): ChartResult {
+    return {
+      status: body.status.toUpperCase() as ChartStatus,
+      error: body.error,
```

Comment:
Why don't we map the error into GraphqlError like transformAskDetailResult does?